src/example-types/results.js

```diff
@@ -31,36 +31,34 @@ module.exports = {
 
     let highlight =
       _.getOr(true, 'highlight', context) && schema.elasticsearch.highlight
-    let inlineAliases = _.getOr(
-      {},
-      'elasticsearch.highlight.inlineAliases',
-      schema
-    )
+    let inlineAliases = _.getOr({}, 'inlineAliases', highlight)
+    let showOtherMatches = _.getOr(false, 'showOtherMatches', context)
 
     if (highlight) {
-      // Only take the fields that matter to highlighting which are the inline, inlineAliases and additionalFields sections
-      let schemaHighlightFields = _.flow(
+      // Convert the highlight fields from array to an object map
+      let fields = _.flow(
         _.pick(['inline', 'additionalFields']),
         _.values,
         _.flatten,
-        _.concat(_.keys(inlineAliases)),
+        _.concat(_.values(inlineAliases)),
         _.uniq(),
-        // intersect with context.include so we only highlight fields we specified in the context if showOtherMatches is set to false
-        fields =>
-          _.getOr(false, 'showOtherMatches', context)
-            ? fields
-            : _.intersection(context.include, fields),
-        // concat the inlineAliases KEYS so they are part of the highlight.fields object so we highlight on them in the ES response
-        _.concat(_.values(inlineAliases))
-      )(schema.elasticsearch.highlight)
+        arrayToHighlightsFieldMap,
+        filtered =>
+          showOtherMatches
+            ? filtered
+            : _.pick(
```

Comment:
pretty sure `_.pick` won't include fields that are missing - meaning you don't need `_.intersection` with `_.keys(filtered))` anymore.